d2compiler/compile.go

```diff
@@ -836,6 +834,19 @@ func (c *compiler) compileArrowheads(edge *d2graph.Edge, f *d2ir.Field) {
 	}
 }
 
+func (c *compiler) nullify(g *d2graph.Graph) {
```

Comment:
I'd prefer to null as it comes in rather then at the end. As, after a null, the object is gone. So if you try to create it again, it shouldn't assume all properties from before the null, only what's added after the null.

Reply:
i first tried to null in the IR, then at compiler as it reads fields, and finally arrived at doing it at the end.

the implicit deletion is hard to do otherwise:

implicit edge
```d2
a -> b
b: null
```

you can only know if the edge should be deleted at the end, if the final ref to one of the endpoints was null.

implicit children
```d2
a.b.c
a.b: null
```

`a.b.c` only exists if `a.b` wasn't nulled somewhere.

> after a null, the object is gone. So if you try to create it again, it shouldn't assume all properties from before the null, only what's added after the null.

oh interesting. yeah i can see that. i can also see the flip side: ordering correctly is annoying. like it's not obvious D2 would evaluate an inner scope's underscore null setting first. will think on it a bit but may just keep this for now.

Reply:
actually, i guess the case can be made that for the implicit edge example, when `b` reinitializes, it's a completely new `b` that doesn't inherit the previous `b`'s connections. likewise with the children

Reply:
yeah ur right, that's better.